### PR TITLE
API-8379 add info about new reports

### PR DESCRIPTION
**Pull request by MaciejSzaflik**
## 🚀 Links

- [Feature branch](https://developers.labs.livechat.com/docs/feature/API-8379-add-ranking-and-ratings-reports)
- [Jira](https://livechatinc.atlassian.net/browse/API-8379)

## 📓 Description

Added information about two new reports in reports-api: ratings and ranking.
Reports 'description' from here: [old documentation](https://github.com/livechat/livechat-docs/blob/master/livechat2/rest-api/docs/chatReports.md#get-ratings-report) 
Added sample responses (without header containing request)
I'm not sure if this 'score' in ranking should be explained?

### Content

- New content
- Proofreading/content fixes

### CODE REVIEW

File: content/data-reporting/reports-api/v3.4/index.mdx [hunk 1]
```diff
@@ -289,6 +291,118 @@ https://api.livechatinc.com/v3.4/reports/chats/total_chats \
 </Code>
 </Section>
 
+<Section>
+<Text>
+
+### Ratings
+
+Shows how many chats were rated and how they have been rated during specified period.
```
Comment (ale-kacperczyk) [inline content/data-reporting/reports-api/v3.4/index.mdx:299]:
```suggestion
Shows the number of rated chats along with their ratings during a specified period of time.
```

File: content/data-reporting/changelog/index.mdx [hunk 2]
```diff
@@ -11,6 +11,11 @@ This document is the record of all the changes in [**the Reports API**](/data-re
 
 ## [v3.4] - Developer preview
 
+### Chats
+
+- There's a new method, [**Ratings**](/data-reporting/reports-api/v3.4/#ratings), which shows how many chats were rated and how they have been rated during specified period.
```
Comment (ale-kacperczyk) [inline content/data-reporting/changelog/index.mdx:16]:
```suggestion
- There's a new method, [**Ratings**](/data-reporting/reports-api/v3.4/#ratings), which shows the number of rated chats along with their ratings during a specified period of time.
```

File: payloads/data-reporting/v3.4/reports-api/responses/chats/ratings.json [hunk 3]
```diff
@@ -0,0 +1,19 @@
+{
```
Comment (ale-kacperczyk) [inline payloads/data-reporting/v3.4/reports-api/responses/chats/ratings.json:1]: I got the following response:

`{
    "name": "ratings-report",
    "request": {
        "distribution": "day",
        "timezone": "Europe/Warsaw",
        "filters": {
            "from": "2021-06-23T02:00:00+02:00",
            "to": "2021-06-28T01:59:59+02:00"
        }
    },
    "total": 0,
    "records": {
        "2021-06-23": {},
        "2021-06-24": {},
        "2021-06-25": {},
        "2021-06-26": {},
        "2021-06-27": {},
        "2021-06-28": {}
    }
}`

Comment (ale-kacperczyk) [reply payloads/data-reporting/v3.4/reports-api/responses/chats/ratings.json:1]: My chats were not rated, so nevermind, but I also got the `request` object

File: content/data-reporting/reports-api/v3.4/index.mdx [hunk 4]
```diff
@@ -289,6 +291,118 @@ https://api.livechatinc.com/v3.4/reports/chats/total_chats \
...
+
+See [available parameters and filters](#available-parameters-and-filters).
+
+#### Response
+
+| Field                    | Notes                                                    |
+| ------------------------ | -------------------------------------------------------- |
+| `ranking`                | Contains the `agent` objects.                            |
+| `ranking.agent.name`     | Name of `agent`                                          |
+| `ranking.agent.total`    | The total number of agent's chats that period.           |
```
Comment (ale-kacperczyk) [inline content/data-reporting/reports-api/v3.4/index.mdx:375]:
```suggestion
| `ranking.agent.total`    | The total number of chats an agent had in the specified period.           |
```

File: content/data-reporting/reports-api/v3.4/index.mdx [hunk 5]
```diff
@@ -289,6 +291,118 @@ https://api.livechatinc.com/v3.4/reports/chats/total_chats \
...
+| `ranking.agent.good`     | The number of agent's chats rated `good`                 |
+| `ranking.agent.bad`      | The number of agent's chats rated `good`                 |
+| `ranking.agent.score`    | Score of agent                                           |
```
Comment (ale-kacperczyk) [inline content/data-reporting/reports-api/v3.4/index.mdx:378]:
```suggestion
| `ranking.agent.score`    | The agent's score. [Read more about how we calculate agent rankings...](https://www.livechat.com/help/agent-rankings/)                                    |
```

File: payloads/data-reporting/v3.4/reports-api/responses/chats/ratings.json (hunk 3 above)
Comment (ale-kacperczyk) [reply payloads/data-reporting/v3.4/reports-api/responses/chats/ratings.json:1]: Same for `/ratings`

File: payloads/data-reporting/v3.4/reports-api/responses/chats/ranking.json [hunk 6]
```diff
@@ -0,0 +1,19 @@
+{
+    "name": "ranking-report",
+    "ranking": [
+        {
+            "name": "test2@livechatinc.com",
```
Comment (ale-kacperczyk) [inline payloads/data-reporting/v3.4/reports-api/responses/chats/ranking.json:5]:
```suggestion
            "name": "smith@example.com",
```

File: payloads/data-reporting/v3.4/reports-api/responses/chats/ranking.json [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+            "total": 27,
+            "good": 14,
+            "bad": 13,
+            "score": 0.309597072108088
+        },
+        {
+            "name": "test1@livechat.com",
```
Comment (ale-kacperczyk) [inline payloads/data-reporting/v3.4/reports-api/responses/chats/ranking.json:12]:
```suggestion
            "name": "agent@example.com",
```

File: content/data-reporting/reports-api/v3.4/index.mdx [hunk 8]
```diff
@@ -289,6 +291,118 @@ https://api.livechatinc.com/v3.4/reports/chats/total_chats \
...
+
+</Code>
+</Section>
+
+<Section>
+<Text>
+
+### Ranking
+
+Shows the ratio of good to bad ratings for each operator.
```
Comment (ale-kacperczyk) [inline content/data-reporting/reports-api/v3.4/index.mdx:355]:
```suggestion
Shows the ratio of good to bad ratings for each agent within a license.
```

File: content/data-reporting/changelog/index.mdx [hunk 9]
```diff
@@ -11,6 +11,11 @@ This document is the record of all the changes in [**the Reports API**](/data-re
...
+- There's a new method, [**Ranking**](/data-reporting/reports-api/v3.4/#ranking), which shows the ratio of good to bad ratings for each operator durning specified period.
```
Comment (ale-kacperczyk) [inline content/data-reporting/changelog/index.mdx:17]:
```suggestion
- There's a new method, [**Ranking**](/data-reporting/reports-api/v3.4/#ranking), which shows the ratio of good to bad ratings for each agent within a license.
```

File: payloads/data-reporting/v3.4/reports-api/responses/chats/ranking.json (hunk 6 above)
Comment (ale-kacperczyk) [reply payloads/data-reporting/v3.4/reports-api/responses/chats/ranking.json:5]: You can find dummy data in `content/default-credentials.md` in the repo

File: content/data-reporting/reports-api/v3.4/index.mdx [hunk 10]
```diff
@@ -289,6 +291,118 @@ https://api.livechatinc.com/v3.4/reports/chats/total_chats \
...
+#### Request
+
+See [available parameters and filters](#available-parameters-and-filters).
+
+#### Response
+
+| Field                    | Notes                                                    |
+| ------------------------ | -------------------------------------------------------- |
+| `ranking`                | Contains the `agent` objects.                            |
+| `ranking.agent.name`     | Name of `agent`                                          |
```
Comment (ale-kacperczyk) [inline content/data-reporting/reports-api/v3.4/index.mdx:374]:
```suggestion
| `ranking.agent.name`     | The agent's name.                                          |
```

File: content/data-reporting/reports-api/v3.4/index.mdx [hunk 11]
```diff
@@ -289,6 +291,118 @@ https://api.livechatinc.com/v3.4/reports/chats/total_chats \
...
+| `ranking.agent.total`    | The total number of agent's chats that period.           |
+| `ranking.agent.good`     | The number of agent's chats rated `good`                 |
```
Comment (ale-kacperczyk) [inline content/data-reporting/reports-api/v3.4/index.mdx:376]:
```suggestion
| `ranking.agent.good`     | The number of an agent's chats rated `good`                 |
```

File: content/data-reporting/reports-api/v3.4/index.mdx [hunk 12]
```diff
@@ -289,6 +291,117 @@ https://api.livechatinc.com/v3.4/reports/chats/total_chats \
...
+
+See [available parameters and filters](#available-parameters-and-filters).
+
+#### Response
+
+| Field                    | Notes                                                                                                                  |
+| ------------------------ | ---------------------------------------------------------------------------------------------------------------------- |
+| `ranking`                | Contains the `agent` objects.                                                                                          |
+| `ranking.agent.total`    | The total number of chats an agent had in the specified period.                                                        |
+| `ranking.agent.good`     | The number of an agent's chats rated `good`                                                                            |
```
Comment (ale-kacperczyk) [inline content/data-reporting/reports-api/v3.4/index.mdx:375]:
```suggestion
| `ranking.agent.good`     | The number of an agent's chats rated `good`.                                                                            |
```

File: content/data-reporting/reports-api/v3.4/index.mdx [hunk 13]
```diff
@@ -289,6 +291,117 @@ https://api.livechatinc.com/v3.4/reports/chats/total_chats \
...
+| `ranking.agent.bad`      | The number of an agent's chats rated `bad`                                                                             |
```
Comment (ale-kacperczyk) [inline content/data-reporting/reports-api/v3.4/index.mdx:376]:
```suggestion
| `ranking.agent.bad`      | The number of an agent's chats rated `bad`.                                                                             |
```

**Review by ale-kacperczyk — APPROVED** [review]
Two tiny things I didn't notice last time, but apart from that, it's all 👍🏻

File: content/data-reporting/reports-api/v3.4/index.mdx [hunk 14]
```diff
@@ -289,6 +291,117 @@ https://api.livechatinc.com/v3.4/reports/chats/total_chats \
...
+| `ranking.agent.score`    | The agent's score. [Read more about how we calculate agent rankings...](https://www.livechat.com/help/agent-rankings/) |
+
+</Text>
+<Code>
+<CodeSample path={'REQUEST'}>
+<Sample>
+
+```shell
+curl -X POST \
+https://api.livechatinc.com/v3.4/reports/chats/ratings \
```
Comment (janowsiany) [inline content/data-reporting/reports-api/v3.4/index.mdx:386]: you mean ranking 😄

File: content/data-reporting/reports-api/v3.4/index.mdx [hunk 15]
```diff
@@ -289,6 +291,117 @@ https://api.livechatinc.com/v3.4/reports/chats/total_chats \
...
+  -H 'Content-Type: application/json' \
+  -H 'Authorization: Bearer <your_access_token>' \
+  -d '{
+    "distribution": "day",
+    "filters": {
+      "from": "2021-06-23T00:00:00-00:00",
+      "to": "2021-06-27T23:59:59-00:00"
```
Comment (janowsiany) [inline content/data-reporting/reports-api/v3.4/index.mdx:337]: can we use the same date range as in other reports?

**Review by janowsiany — CHANGES_REQUESTED** [review]

File: content/data-reporting/reports-api/v3.4/index.mdx (hunk 14 above)
Comment (MaciejSzaflik) [reply content/data-reporting/reports-api/v3.4/index.mdx]: 🤦

File: content/data-reporting/reports-api/v3.4/index.mdx (hunk 15 above)
Comment (MaciejSzaflik) [reply content/data-reporting/reports-api/v3.4/index.mdx]: changed, along with response

**Review by janowsiany — APPROVED** [review]

**Pull request merged**